get_sample_covariates.py

```diff
@@ -98,29 +85,37 @@ def main(
     init_batch()
 
     # sex
-    # check if files exist
-    try:
-        # TOB sex info from Vlad's metadata file
-        tob_meta = pd.read_csv(tob_sex_file_path)
-        # BioHEART sex info from Hope's Somalier stand alone run
-        bioheart_meta = pd.read_csv(bioheart_sex_file_path, sep="\t")
-    except FileNotFoundError as e:
-        print(f"Error: File not found - {e}")
-        sys.exit(1)
-    # extract sex for TOB
-    # remove samples with ambiguous sex inference
-    tob_meta = tob_meta[tob_meta['sex_karyotype'].isin(["XX", "XY"])]
-    # encode sex as 1,2 instead
-    tob_meta['sex'] = tob_meta['sex_karyotype'].replace('XY', '1')
-    tob_meta['sex'] = tob_meta['sex'].replace('XX', '2')
-    # rename s as sample id to match bioheart file
-    tob_meta['sample_id'] = tob_meta['new_CPG_id']
-    tob_sex = tob_meta.loc[:, ["sample_id", "sex"]]
-
-    # extract sex for BioHEART
-    bioheart_sex = bioheart_meta.loc[:, ["sample_id", "sex"]]
+    # option 1: separate files
+    # tob
+    tob_sample_qc_ht = hl.read_table(tob_sampleqc_file_path)
+    # convert to pandas
+    tob_sample_qc_df = tob_sample_qc_ht.to_pandas()
```

Comment:
Why read the table with hail and then convert to pandas, rather than just read with pandas to start? @MattWellie anything I'm missing here?

Reply:
ooo did not know you could do that?!

Reply:
From memory the input file is in a proprietary Hail format, so a read + convert with Hail was required, it's not a simple T/CSV

Reply:
Naive try in a notebook didn't succeed, but happy to try things?

<img width="1008" alt="Screenshot 2024-09-25 at 11 57 59 AM" src="https://github.com/user-attachments/assets/4e1455e0-56cd-4de8-8bc6-e8d2abbc9569">